### PR TITLE
Allow to drag and drop all elements from tree view to diagram

### PR DESCRIPTION
**Pull request by amolenaar**
### PR Type
What kind of change does this PR introduce?

- [ ] Bug fix
- [x] Feature
- [ ] Chore (refactoring, formatting, local variables, other cleanup)
- [ ] Documentation content changes

### What is the current behavior?

Only classes, packages can be dragged from tree view to diagram.

Issue Number: #648 

### What is the new behavior?

Model elements can be dragged from tree view to diagram.

- [x] Relationships will connect to their ends
- [x] Relationships can only be dropped if both ends are present on the diagram, otherwise a notification will be shown

### Does this PR introduce a breaking change?
- [x] Yes
- [ ] No

### Other information

Covered "simple" elements and relationships, complex relationships (Association, Extension, Connector, Message) and Proxy port.

I think that's all.

### CODE REVIEW

File: gaphor/ui/diagrampage.py [hunk 1]
```diff
@@ -340,36 +336,20 @@ def _on_drag_data_received(self, view, context, x, y, data, info, time):
...
-                    element, (UML.Classifier, UML.Package, UML.Property)
-                ) or isinstance(element, UML.Association):
-                    self.event_manager.handle(
-                        Notification(
-                            gettext(
-                                "Drag to diagram is (temporarily) limited to Classifiers, Packages, and Properties, not {type}."
-                            ).format(type=type(element).__name__)
+                x, y = view.matrix.inverse().transform_point(x, y)
+                with Transaction(self.event_manager):
+                    if item := drop(element, self.diagram, x, y):
```
Comment (amolenaar) [inline gaphor/ui/diagrampage.py:341]: This is the focus point of this PR.

**Review by danyeaw — APPROVED** [review]
This looks really good!

File: gaphor/diagram/tools/dnd.py [hunk 2]
```diff
@@ -17,5 +17,6 @@ def on_drop(target, tool_name, x, y, modeling_language, event_manager):
     tool_def = get_tool_def(modeling_language, tool_name)
     with Transaction(event_manager):
         create_item(view, tool_def.item_factory, x, y)
-
+    # TODO: open editor
```
[the remark this answers is not in the document]
Comment (amolenaar) [reply gaphor/diagram/tools/dnd.py]: I removed them.

**Comment by amolenaar** [comment]
I added a few more things: a section in the docs, and a rule in the model-consistency test.

**Review by danyeaw — APPROVED** [review]

**Pull request merged**